File: module-1-number-theory/03_GCD_and_LCM_part1/01_CEQU_CrucialEquation/Main.java

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
	public static StringBuilder output = new StringBuilder();

	public static void main(String[] args) {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		final int cases;
		try {
			cases = Integer.parseInt(br.readLine().trim());
			Solver solver = new Solver();
			for (int i = 0; i < cases; i++) {				
				solver.solve(i, br.readLine());
			}

		} catch (IOException e) {
			e.printStackTrace();
		}
		// Print the final output
		System.out.println(output.toString().trim());
	}

}
class Solver {
	public int gcd(int a, int b) {
		if (b == 0) {
			return a;
		}
		return gcd(b, a % b);
	}
    
    
    public void solve(int caseNumber, String input) {					
		String[] inputs = input.split(" ");
        int a = Integer.parseInt(inputs[0]);
        int b = Integer.parseInt(inputs[1]);
        int c = Integer.parseInt(inputs[2]);

        int g = gcd(a, b);
        int l = c % g;

        Main.output.append("Case ").append(caseNumber + 1).append(": ");
        if(l == 0) {
            Main.output.append("Yes").append("\n");
        } else {
            Main.output.append("No").append("\n");
        }
	}
}

File: module-1-number-theory/03_GCD_and_LCM_part1/01_CEQU_CrucialEquation/Main.py
#!/usr/bin/env python3
import sys

input = sys.stdin.readline 

OFFLINE_TEST = False


def log(*args):
    print("\033[36m", *args, "\033[0m", file=sys.stderr)


def solve(*args):
    # screen input
    if OFFLINE_TEST:
        log("----- solving ------")
        log(*args)
        log("--------------------")
    return solve_(*args)

def gcd(a,b):
    if b == 0:
        return a
    return gcd(b,a%b)


def solve_(a,b,c):
    g = gcd(a,b)
    l = c%g
    if l == 0:
        return True
    return False

def main():
    for case_num in range(int(input())):
        a,b,c = map(int, input().split())
        res = solve(a,b,c) 
        if res == True:
            print(f"Case {case_num+1}: Yes")
        else:
            print(f"Case {case_num+1}: No")

if __name__ == "__main__":
    main()